src/main/java/com/capitalone/dashboard/collector/DefaultWhiteSourceClient.java

```diff
@@ -0,0 +1,316 @@
+            return orgName;
+        } catch (Exception e) {
+            LOG.error("Exception occurred while calling getOrgDetails " + e.getStackTrace());
+        }
+        return null;
+    }
+
+    @Override
+    public List<WhiteSourceProduct> getProducts(String instanceUrl, String orgToken) {
+        List<WhiteSourceProduct> whiteSourceProducts = new ArrayList<>();
+        try {
+            JSONObject jsonObject = makeRestCall(getApiBaseUrl(instanceUrl), Constants.RequestType.getAllProducts, orgToken, null, null);
+            if (Objects.isNull(jsonObject)) return new ArrayList<>();
+            JSONArray jsonArray = (JSONArray) jsonObject.get(Constants.PRODUCTS);
+            if (CollectionUtils.isEmpty(jsonArray)) return new ArrayList<>();
+            for (Object product : jsonArray) {
+                JSONObject wsProduct = (JSONObject) product;
+                WhiteSourceProduct whiteSourceProduct = new WhiteSourceProduct();
+                whiteSourceProduct.setProductId(getLongValue(wsProduct, Constants.PRODUCT_ID));
+                whiteSourceProduct.setProductName(getStringValue(wsProduct, Constants.PRODUCT_NAME));
+                whiteSourceProduct.setProductToken(getStringValue(wsProduct, Constants.PRODUCT_TOKEN));
+                whiteSourceProducts.add(whiteSourceProduct);
+            }
+        } catch (Exception e) {
+            LOG.error("Exception occurred while retrieving getAllProducts " + e.getMessage());
+        }
+        return whiteSourceProducts;
+    }
+
+    @Override
+    public List<WhiteSourceComponent> getAllProjectsForProduct(String instanceUrl, WhiteSourceProduct product, String orgToken, String orgName) {
+        List<WhiteSourceComponent> whiteSourceProjects = new ArrayList<>();
+        try {
+            JSONObject jsonObject = makeRestCall(getApiBaseUrl(instanceUrl), Constants.RequestType.getAllProjects, orgToken, product.getProductToken(), null);
+            if (Objects.isNull(jsonObject)) return new ArrayList<>();
+            JSONArray jsonArray = (JSONArray) jsonObject.get(Constants.PROJECTS);
+            if (jsonArray == null) return new ArrayList<>();
+            for (Object project : jsonArray) {
+                JSONObject wsProject = (JSONObject) project;
+                WhiteSourceComponent whiteSourceProject = new WhiteSourceComponent();
+                whiteSourceProject.setProjectName(getStringValue(wsProject, Constants.PROJECT_NAME));
+                whiteSourceProject.setProjectToken(getStringValue(wsProject, Constants.PROJECT_TOKEN));
+                whiteSourceProject.setProductToken(product.getProductToken());
+                whiteSourceProject.setProductName(product.getProductName());
+                whiteSourceProject.setOrgName(orgName);
+                whiteSourceProjects.add(whiteSourceProject);
+            }
+        } catch (Exception e) {
+            LOG.error("Exception occurred while retrieving getAllProducts " + e.getMessage());
```

Comment:
Same as above comments, don't let the logic continue, skip to the next org.
I think the same comment applies to most of the calls below.

Reply:
this should not skip to next org unless all the products are iterated. We can add logging for product name for exceptions.